src/backend/distributed/commands/alter_table.c

```diff
@@ -1250,6 +1256,82 @@ CreateCitusTableLike(TableConversionState *con)
 }
 
 
+/*
+ * ErrorForUnsupportedCascadeObjects gets oid of a relation, finds the objects
+ * that dropping this relation cascades into and errors if there are any extensions
+ * that would be dropped.
+ */
+static void
+ErrorForUnsupportedCascadeObjects(Oid relationId)
+{
+	HASHCTL info;
+	memset(&info, 0, sizeof(info));
+	info.keysize = sizeof(Oid);
+	info.entrysize = sizeof(Oid);
+	info.hash = oid_hash;
+	uint32 hashFlags = (HASH_ELEM | HASH_FUNCTION);
+	HTAB *nodeMap = hash_create("object dependency map (oid)", 64, &info, hashFlags);
+
+	TraverseCascadeObjectsGraph(RelationRelationId, relationId, nodeMap);
+}
+
+
+/*
+ * TraverseCascadeObjectsGraph walks through the objects that are dependant of the
+ * object with object id and errors if it finds any extensions.
+ * 
+ * Extension dependency is different than the rest. If an object depends on an extension
+ * dropping the object would drop the extension too.
+ * So we check with IsObjectAddressOwnedByExtension function.
+ */
+static void
+TraverseCascadeObjectsGraph(Oid classId, Oid objectId, HTAB *nodeMap)
+{
+	bool found = false;
+	hash_search(nodeMap, &objectId, HASH_ENTER, &found);
+	
+	if (found)
+	{
+		return;
+	}
+
+	ObjectAddress objectAddress = { 0 };
+	ObjectAddressSet(objectAddress, classId, objectId);
+
+	if (IsObjectAddressOwnedByExtension(&objectAddress, NULL))
+	{
+		ereport(ERROR, (errmsg("cannot alter table because an extension depends on it")));
```

Comment:
I would not have guessed that this function would not return any errors. Can you either (a) rename the function, or (b) aggregate and return all the unsupported dependencies in `TraverseCascadeObjectsGraph` and report the errors in `ErrorIf..` function?

I'd like option (b) more.

Reply:
I need to agree with Hanefi on this one. Instead of throwing the error from here, could we return a bool, either
 - `true` the graph explored has an extension
 -  `false` the graph explored doesn't have an extension
We would stop traversing (and return `true`) once we find the condition we error on here.

This would go well with renaming the function to something descriptive like, `ObjectDropCascadesToExtension`. Then the error could be thrown from `ErrorForUnsupportedCascadeObjects`, which makes it  easier to reason about.